lib/ocpp/common/websocket/websocket_libwebsockets.cpp

```diff
@@ -1298,4 +1295,25 @@ int WebsocketTlsTPM::process_callback(void* wsi_ptr, int callback_reason, void*
     return 0;
 }
 
+void WebsocketTlsTPM::push_deferred_callback(std::function<void()> callback) {
+    std::scoped_lock tmp_lock(this->deferred_callback_mutex);
+    this->deferred_callback_queue.push(callback);
+    this->deferred_callback_cv.notify_one();
+}
+
+void WebsocketTlsTPM::handle_deferred_callback_queue() {
+    while (true) {
```

Comment:
Does it look OK to always execute this thread? Maybe when we close the connection we should somehow have a way to end the thread and also join it in the destructor, not good practice to leave a thread running in the background.

Reply:
Ah yes you are right. I am relatively new to threads and expected it to be handled cleanly by the thread destructor but that is not the case. I will add this.

Reply:
I think it should be fine to leave the thread alive while we are closed right? Otherwise we might open up issues with callbacks not being called etc.

Reply:
That is correct. The behavior should be as following: 
1. Have the thread always run if there are actions in the queue or if there is a 'live' connection (conn_data.is_valid() or something along those lines)
2. If the connection is closed, and the queue is not empty, still keep the thread running until the queue is depleted
3. If a 'reconnect' is initiated and the thread is already running do NOT start a new thread but re-use the existing thread

Reply:
however maybe we want to be able to stop this thread when we shutdown completely, so having this use a member variable instead of just "true" (that could potentially be set from the outside to "false") is probably not a bad idea

Reply:
I do that using a break statement instead of checking it in the while loop. I can add it/move it to the while as well if you prefer?

Reply:
Ah you're right, must have missed that earlier somehow. No than it's fine I would say